airbyte-integrations/connectors/source-zoho-crm/source_zoho_crm/api.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from types import MappingProxyType
+from typing import Any, List, Mapping, MutableMapping, Tuple
+from urllib.parse import urlsplit, urlunsplit
+
+import requests
+
+from .auth import ZohoOauth2Authenticator
+
+logger = logging.getLogger(__name__)
+
+
+class ZohoAPI:
+    _DC_REGION_TO_ACCESS_URL = MappingProxyType(
+        {
+            "US": "https://accounts.zoho.com",
+            "AU": "https://accounts.zoho.com.au",
+            "EU": "https://accounts.zoho.eu",
+            "IN": "https://accounts.zoho.in",
+            "CN": "https://accounts.zoho.com.cn",
+            "JP": "https://accounts.zoho.jp",
+        }
+    )
+    _DC_REGION_TO_API_URL = MappingProxyType(
+        {
+            "US": "https://zohoapis.com",
+            "AU": "https://zohoapis.com.au",
+            "EU": "https://zohoapis.eu",
+            "IN": "https://zohoapis.in",
+            "CN": "https://zohoapis.com.cn",
+            "JP": "https://zohoapis.jp",
+        }
+    )
+    _API_ENV_TO_URL_PREFIX = MappingProxyType({"production": "", "developer": "developer", "sandbox": "sandbox"})
+    _CONCURRENCY_API_LIMITS = MappingProxyType({"Free": 5, "Standard": 10, "Professional": 15, "Enterprise": 20, "Ultimate": 25})
+
+    def __init__(self, config: Mapping[str, Any]):
+        self.config = config
+        self._authenticator = None
+
+    @property
+    def authenticator(self) -> ZohoOauth2Authenticator:
```

Comment:
just a question: is there a reason to lazy instantiate this instead of just creating it in the `__init__` method? (maybe putting this logic in a helper method `create_authenticator` or something?

Reply:
As a rule, I try to make `__init__` methods as lightweight as possible.
A helper method would give no guarantees it would be called once and a stream would work with the only authenticator instance (what is important in context of number of API requests made). If you insist, I'd better move it to `__init__`

Reply:
ok sounds good as is. i was just curious.